cpp/include/cudf_test/column_utilities.hpp

```diff
@@ -44,36 +60,45 @@ void expect_column_properties_equal(cudf::column_view const& lhs, cudf::column_v
  *
  * @param lhs The first column
  * @param rhs The second column
+ * @param verbosity Level of debug output verbosity
+ *
+ * @returns True if the column properties are equivalent, false otherwise
  */
-void expect_column_properties_equivalent(cudf::column_view const& lhs,
-                                         cudf::column_view const& rhs);
+bool expect_column_properties_equivalent(
+  cudf::column_view const& lhs,
+  cudf::column_view const& rhs,
+  debug_output_level verbosity = debug_output_level::FIRST_ERROR);
 
 /**
  * @brief Verifies the element-wise equality of two columns.
  *
  * Treats null elements as equivalent.
  *
- * @param lhs                   The first column
- * @param rhs                   The second column
- * @param print_all_differences If true display all differences
+ * @param lhs The first column
+ * @param rhs The second column
+ * @param verbosity Level of debug output verbosity
+ *
+ * @returns True if the columns (and their properties) are equal, false otherwise
  */
-void expect_columns_equal(cudf::column_view const& lhs,
+bool expect_columns_equal(cudf::column_view const& lhs,
                           cudf::column_view const& rhs,
-                          bool print_all_differences = false);
+                          debug_output_level verbosity = debug_output_level::FIRST_ERROR);
 
 /**
  * @brief Verifies the element-wise equivalence of two columns.
  *
  * Uses machine epsilon to compare floating point types.
  * Treats null elements as equivalent.
```

Comment:
How about `expect_columns_equal`? Does it also uses machine epsilon to compare floating points? If so, can you add this comment to that function please?

Personally, I want to have an API to add a user-specified epsilon instead (with default value is machine epsilon). This would be helpful to compare the computation results of large datasets, which may have very large round-of errors. Anyway, such API can be added later in a separate PR when needed.

Reply:
expect_columns_equal() does not. It ends up using `thrust::equal()` element by element.